### ISSUE TITLE
esbuild plugin: timeout after 10 second on `deno task build`

### ISSUE THREAD

**Issue by scarf005** [lumeland/lume#591]
### Version

2.1.3

### Platform

ubuntu

### What steps will reproduce the bug?

1. Create project with following files: `deno.json`, `_config.ts`, `main.tsx` 

`deno.json`
```json
{
	"imports": {
		"lume/": "https://deno.land/x/lume@v2.1.3/"
	},
	"tasks": {
		"lume": "echo \"import 'lume/cli.ts'\" | deno run -A -",
		"build": "deno task lume",
		"serve": "deno task lume -s"
	},
	"compilerOptions": {
		"exactOptionalPropertyTypes": true,
		"types": [
			"lume/types.ts"
		],
		"jsx": "precompile",
		"jsxImportSource": "https://esm.sh/preact@10.19.3"
	}
}
```

`_config.ts`
```ts
import lume from "lume/mod.ts"
import esbuild from "lume/plugins/esbuild.ts"

const site = lume()

site.use(esbuild({
	extensions: [".tsx"],
	options: { minify: false, bundle: false, splitting: true },
}))

export default site
```

`main.tsx`
```
import { render } from "https://esm.sh/preact@10.19.3"

const App = () => <h1>Example Page</h1>

render(<App />, document.getElementById("root")!)
```

2. run `deno task build`.

### How often does it reproduce? Is there a required condition?

it always happens.

### What is the expected behavior?

after building the project must stop immediately.

### What do you see instead?

```
$ deno task build
Task build deno task lume
Task lume echo "import 'lume/cli.ts'" | deno run -A -
Loading config file file:///home/scarf/repo/etc/esbuild-issue/_config.ts
🔥 /main.js <- /main.tsx
🍾 Site built into ./_site
  1 files generated in 0.01 seconds
WARN After waiting 10 seconds, there are some timers that avoid ending the process.
WARN They have been forcibly closed.
```

The plugin hangs and gets killed after 10 seconds.

### Additional information

it might be that `esbuild.stop()` isn't called?

**Comment by oscarotero** [comment]
@scarf005 It was removed here https://github.com/lumeland/lume/commit/bff5800be2414b80f6288e27bc8d8e0be832117a#diff-019b7b01afa861216d127d59201e36bc7b5ccbb44be90f521af6f044808b09a1

I don't remember exactly the reason, but I got some errors importing `stop` so I though it wasn't needed anymore (`stop` only is required in Deno, in Node it works automatically).

**Comment by scarf005** [comment]
As a workaround, adding this to `_config.ts` works:

```ts
import { stop } from "lume/deps/esbuild.ts"

const isWatch = Deno.args.some((arg) => ["-s", "--serve", "-w", "--watch"].includes(arg))
if (!isWatch) addEventListener("lume:afterBuild", stop)
```

**Comment by oscarotero** [comment]
great. 
Could you try if this happens with `splitting: false`?
The plugins works differently depending on this option in the esbuild settings.

**Comment by scarf005** [comment]
changing the value of `splitting` doesn't change the result, probably due to https://esbuild.github.io/getting-started/#deno

**Comment by oscarotero** [comment]
fixed in v2.1.4
